src/Stratis.Bitcoin.IntegrationTests.Common/TestHelper.cs

```diff
@@ -33,20 +33,42 @@ public static void WaitLoop(Func<bool> act, string failureReason = "Unknown Reas
 
         public static bool AreNodesSynced(CoreNode node1, CoreNode node2)
         {
-            if (node1.FullNode.Chain.Tip.HashBlock != node2.FullNode.Chain.Tip.HashBlock) return false;
-            if (node1.FullNode.ChainBehaviorState.ConsensusTip.HashBlock != node2.FullNode.ChainBehaviorState.ConsensusTip.HashBlock) return false;
-            if (node1.FullNode.GetBlockStoreTip().HashBlock != node2.FullNode.GetBlockStoreTip().HashBlock) return false;
-            if (node1.FullNode.MempoolManager().InfoAll().Count != node2.FullNode.MempoolManager().InfoAll().Count) return false;
-            if (node1.FullNode.WalletManager().WalletTipHash != node2.FullNode.WalletManager().WalletTipHash) return false;
-            if (node1.CreateRPCClient().GetBestBlockHash() != node2.CreateRPCClient().GetBestBlockHash()) return false;
+            if (node1.FullNode.Chain.Tip.HashBlock != node2.FullNode.Chain.Tip.HashBlock)
+                return false;
+
+            if (node1.FullNode.ChainBehaviorState.ConsensusTip.HashBlock != node2.FullNode.ChainBehaviorState.ConsensusTip.HashBlock)
+                return false;
+
+            if (node1.FullNode.GetBlockStoreTip().HashBlock != node2.FullNode.GetBlockStoreTip().HashBlock)
+                return false;
+
+            if (node1.FullNode.MempoolManager().InfoAll().Count != node2.FullNode.MempoolManager().InfoAll().Count)
+                return false;
+
+            if ((node1.FullNode.WalletManager().Wallets.Count != 0) && (node2.FullNode.WalletManager().Wallets.Count != 0))
+            {
+                if (node1.FullNode.WalletManager().WalletTipHash != node2.FullNode.WalletManager().WalletTipHash)
+                    return false;
+            }
+
+            if (node1.CreateRPCClient().GetBestBlockHash() != node2.CreateRPCClient().GetBestBlockHash())
+                return false;
+
             return true;
         }
 
         public static bool IsNodeSynced(CoreNode node)
         {
-            if (node.FullNode.Chain.Tip.HashBlock != node.FullNode.ChainBehaviorState.ConsensusTip.HashBlock) return false;
-            if (node.FullNode.Chain.Tip.HashBlock != node.FullNode.GetBlockStoreTip().HashBlock) return false;
-            if (node.FullNode.Chain.Tip.HashBlock != node.FullNode.WalletManager().WalletTipHash) return false;
+            if (node.FullNode.Chain.Tip.HashBlock != node.FullNode.ChainBehaviorState.ConsensusTip.HashBlock)
```

Comment:
I think something was missed in the merge, as @ferdeen already fixed this in https://github.com/stratisproject/StratisBitcoinFullNode/commit/dd1056f6ea117c978106d3ec5a12875419e398fc

Reply:
It seems this was missed during the merge 😢

Reply:
All good, at least we found the same problem 👍